src/engraving/layout/layoutmeasure.cpp

```diff
@@ -690,6 +690,9 @@ void LayoutMeasure::getNextMeasure(const LayoutOptions& options, LayoutContext&
                     if (!cr) {
                         continue;
                     }
+                    // Check if requested cross-staff is possible
+                    cr->checkStaffMoveValidity();
```

Comment:
It seems that from now on it will be called much more often than before. Won't it hurt performance?

Reply:
That can change from score to score, but in general we are actually calling this logic _less_ often, because now we call it once per chord:
<img width="647" alt="image" src="https://user-images.githubusercontent.com/93707756/197994509-b9d7c556-166b-4fb9-8a23-930f7004695d.png">
but before we used to call it once per note:
<img width="652" alt="image" src="https://user-images.githubusercontent.com/93707756/197994217-491477cd-0075-411d-a53d-37f0756564ce.png">
Now I've also moved that if condition on the caller side, so we avoid entering the funciton at all if we don't need it. So performance-wise we should be equal or better than before